diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index 3e6189d..13bab57 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -807,6 +807,7 @@ jobs:
         TEST_DB_NAME: ${{ env.MYSQL_DATABASE }}
         TEST_DB_PASSWORD: ${{ env.MYSQL_PASSWORD }}
         TEST_DB_USER: ${{ env.MYSQL_USERNAME }}
+        TEST_DB_HOSTNAME: localhost
         TEST_DB_PORT: ${{ job.services.mysql.ports[3306] }}
         ASHERAH_SERVICE_NAME: service
         ASHERAH_PRODUCT_NAME: product
diff --git a/tests/cross-language/go/encrypt_test.go b/tests/cross-language/go/encrypt_test.go
index d5ad880..2dc9bae 100644
--- a/tests/cross-language/go/encrypt_test.go
+++ b/tests/cross-language/go/encrypt_test.go
@@ -115,9 +115,10 @@ func FeatureContext(s *godog.Suite) {
 // connectionString returns the RDBMS connection string
 func connectionString() string {
 	return fmt.Sprintf(
-		"%s:%s@tcp(localhost:%s)/%s",
+		"%s:%s@tcp(%s:%s)/%s",
 		os.Getenv("TEST_DB_USER"),
 		os.Getenv("TEST_DB_PASSWORD"),
+		os.Getenv("TEST_DB_HOSTNAME"),
 		os.Getenv("TEST_DB_PORT"),
 		os.Getenv("TEST_DB_NAME"))
 }
